toolchain/semantics/semantics_context.cpp

```diff
@@ -502,14 +504,30 @@ auto SemanticsContext::ImplicitAsImpl(SemanticsNodeId value_id,
     // Type doesn't need to change.
     return ImplicitAsKind::Identical;
   }
-
   if (as_type_id == SemanticsTypeId::TypeType) {
-    // TODO: When converting `()` to a type, the result is `() as Type`.
-    // Right now there is no tuple value support.
-
+    if (value.kind() == SemanticsNodeKind::TupleValue) {
+      auto tuple_block_id = value.GetAsTupleValue();
+      llvm::SmallVector<SemanticsTypeId> type_ids;
+      // If it is empty tuple type, we don't fetch anything
+      if (tuple_block_id != SemanticsNodeBlockId::Empty) {
+        auto tuple_block = semantics_ir_->GetNodeBlock(tuple_block_id);
```

Comment:
```suggestion
        const auto &tuple_block = semantics_ir_->GetNodeBlock(tuple_block_id);
```
Avoid making a copy of the vector here.